core/src/main/kotlin/com/alecstrong/sql/psi/core/sqlite_3_35/psi/mixins/AlterTableDropColumnMixin.kt

```diff
@@ -0,0 +1,84 @@
+package com.alecstrong.sql.psi.core.sqlite_3_35.psi.mixins
+
+import com.alecstrong.sql.psi.core.SqlAnnotationHolder
+import com.alecstrong.sql.psi.core.psi.AlterTableApplier
+import com.alecstrong.sql.psi.core.psi.LazyQuery
+import com.alecstrong.sql.psi.core.psi.SqlColumnName
+import com.alecstrong.sql.psi.core.psi.SqlCompositeElement
+import com.alecstrong.sql.psi.core.psi.SqlCompositeElementImpl
+import com.alecstrong.sql.psi.core.psi.SqlCreateIndexStmt
+import com.alecstrong.sql.psi.core.psi.SqlTypes
+import com.alecstrong.sql.psi.core.psi.alterStmt
+import com.alecstrong.sql.psi.core.psi.mixins.ColumnDefMixin
+import com.alecstrong.sql.psi.core.sqlite_3_35.psi.SqliteAlterTableDropColumn
+import com.intellij.lang.ASTNode
+
+internal abstract class AlterTableDropColumnMixin(
+  node: ASTNode
+) : SqlCompositeElementImpl(node),
+  SqliteAlterTableDropColumn,
+  AlterTableApplier {
+  private val columnName
+    get() = children.filterIsInstance<SqlColumnName>().single()
+
+  override fun applyTo(lazyQuery: LazyQuery): LazyQuery {
+    return LazyQuery(
+      tableName = lazyQuery.tableName,
+      query = {
+        lazyQuery.query.copy(
+          columns = lazyQuery.query.columns.filterNot { (it.element as SqlColumnName).textMatches(columnName) }
+        )
+      }
+    )
+  }
+
+  override fun annotate(annotationHolder: SqlAnnotationHolder) {
```

Comment:
I didn't include an exhaustive check for invalid conditions (https://www.sqlite.org/lang_altertable.html#alter_table_drop_column) mostly because I ran out of time to work on this :smile:

Reply:
all good, this is a ton :+1: